sdk/core/core/test/cmocka/test_az_span.c

```diff
@@ -751,6 +751,98 @@ static void az_span_u32toa_overflow_fails(void** state)
   assert_true(az_span_u32toa(buffer, v, &out_span) == AZ_ERROR_INSUFFICIENT_SPAN_SIZE);
 }
 
+static void az_span_atod_test(void** state)
+{
+  (void)state;
+  uint8_t buff[10];
+  az_span buff_span = AZ_SPAN_FROM_BUFFER(buff);
+  az_span remainder = { 0 };
+  remainder = az_span_copy_u8(buff_span, '-');
+  remainder = az_span_copy_u8(remainder, '1');
+  remainder = az_span_copy_u8(remainder, '2');
+  remainder = az_span_copy_u8(remainder, '2');
+  remainder = az_span_copy_u8(remainder, '2');
+  remainder = az_span_copy_u8(remainder, '.');
+  remainder = az_span_copy_u8(remainder, '5');
+  remainder = az_span_copy_u8(remainder, '6');
+  remainder = az_span_copy_u8(remainder, '8');
+  remainder = az_span_copy_u8(remainder, '1');
```

Comment:
nit: Can we inline initialize the `buff` array with these values rather than calling span copy?

Reply:
yeah, but this is a tests, so, no damage at all. Even better to add more testing around

Reply:
I don't think that is the intent of unit tests, but it was a readability nit. Up to you.